pkg/storage/storage_controller.go

```diff
@@ -3,11 +3,14 @@ package storage
 import (
 	"fmt"
 	"strings"
+
+	"zotregistry.io/zot/pkg/metadata"
 )
 
 type StoreController struct {
-	DefaultStore ImageStore
-	SubStore     map[string]ImageStore
+	DefaultStore  ImageStore
+	SubStore      map[string]ImageStore
+	MetadataStore metadata.Store
```

Comment:
The MetadataStore is unrelated to the StoreController. Let's keep it separate.

@rchincha ^^